### PR TITLE
net-vpn/i2p: bump to 0.9.47

### PR DESCRIPTION
**Pull request by tharvik**
* mostly copied from 0.9.45
  * added patch for http://trac.i2p2.i2p/ticket/2770
* drop the old unapplied net-libs/nativebiginteger patches
* dropping the old version as usual

### CODE REVIEW

File: net-vpn/i2p/files/DataHelper-decompress-fix-length-check.patch [hunk 1]
```diff
@@ -0,0 +1,24 @@
+From ceb0749e0d5c3c496b4a91bc3a9d3b5e296fc456 Mon Sep 17 00:00:00 2001
```
Comment (thesamesam) [inline net-vpn/i2p/files/DataHelper-decompress-fix-length-check.patch:1]: Pop the link to the patch at the top (non-I2P link, if you can) for reference.

File: net-vpn/i2p/files/DataHelper-decompress-fix-length-check.patch [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+From: zzz <zzz@mail.i2p>
```
Comment (thesamesam) [inline net-vpn/i2p/files/DataHelper-decompress-fix-length-check.patch:2]: Also rename the patch file to be i2p-0.9.47-..., then refer to it as ${P} in the ebuild as it is upstreamed and should be in the next release.

**Review by thesamesam — COMMENTED** [review]
Minor comments. Can do on merge.

File: net-vpn/i2p/files/DataHelper-decompress-fix-length-check.patch (hunk 2 above)
Comment (tharvik) [reply net-vpn/i2p/files/DataHelper-decompress-fix-length-check.patch]: Good idea, mv'ed

**Comment by thesamesam** [comment]
This looks ready to merge, apart from the fact that sign-offs must be using your real [name](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).

**Comment by tharvik** [comment]
> sign-offs must be using your real name.
Ho, right, fixed.

**Comment by thesamesam** [comment]
```
>>> Downloading 'https://download.i2p2.de/releases/0.9.47/i2psource_0.9.47.tar.bz2'
--2020-09-09 02:33:36--  https://download.i2p2.de/releases/0.9.47/i2psource_0.9.47.tar.bz2
Resolving download.i2p2.de... 193.150.121.26
Connecting to download.i2p2.de|193.150.121.26|:443... connected.
ERROR: cannot verify download.i2p2.de's certificate, issued by 'CN=Let\'s Encrypt Authority X3,O=Let\'s Encrypt,C=US':
  Issued certificate has expired.
To connect to download.i2p2.de insecurely, use `--no-check-certificate'.
!!! Couldn't download 'i2psource_0.9.47.tar.bz2'. Aborting.
^[[31;01m * ^[[39;49;00mFetch failed for 'net-libs/nativebiginteger-0.9.47', Log file:
^[[31;01m * ^[[39;49;00m '/var/tmp/portage/net-libs/nativebiginteger-0.9.47/temp/build.log'
```

**Pull request closed without merging**